src/routes/articles/update-article.e2e.test.js

```diff
@@ -0,0 +1,196 @@
+				},
+				"status": "failed"
+			}
+			return fixtures.api()
+				.patch(`/api/v1/articles/${article.id}`)
+				.set('Authorization', `Bearer ${accessToken}`)
+				.send({ 
+					title : updatedInfo.title,
+					image : updatedInfo.image,
+					article : 123,
+					published : updatedInfo.published
+				})
+				.expect(400, expectedError)
+		})
+		it('should return 400 if image is not a valid url',
+		 async () => {
+				const expectedError = {
+					"error": {
+						"message": "image must be a string"
+					},
+					"status": "failed"
+				}
+				return fixtures.api()
+					.patch(`/api/v1/articles/${article.id}`)
+					.set('Authorization', `Bearer ${accessToken}`)
+					.send({
+						title : updatedInfo.title,
+						image : 123,
+						article : updatedInfo.content,
+						published : updatedInfo.published
+					})
+					.expect(400, expectedError)
+			})
+		it('should return 400 if published is not a boolean',
+			async () => {
+				const expectedError = {
+					"error": {
+						"message": "published must be a boolean"
+					},
+					"status": "failed"
+				}
+				return fixtures.api()
+					.patch(`/api/v1/articles/${article.id}`)
+					.set('Authorization', `Bearer ${accessToken}`)
+					.send({
+						title : updatedInfo.title,
+						image : updatedInfo.image,
+						article : updatedInfo.content,
+						published : 123
+					})
+					.expect(400, expectedError)
+			})
+
+		it('should return 404 if article is not found', async () => 
+			
+			fixtures.api()
+				.patch(`/api/v1/articles/${article.id + 2}`)
+				.set('Authorization', `Bearer ${accessToken}`)
+				.send({
+					title : updatedInfo.title,
+					image : updatedInfo.image,
+					article : updatedInfo.content,
+					published : updatedInfo.published
+					 })
+				.expect(404)
+				.then((res) => {
+					expect(res.body.status).eql('fail')
+					expect(res.body.message).eql('Article does not exist')
+				})
+		)
+
+	})
+	describe('Success', () => {
+		it('should return 200 if article is updated', async () => 
+			fixtures.api()
+				.patch(`/api/v1/articles/${article.id}`)
+				.set('Authorization', `Bearer ${accessToken}`)
+				.send({ 
+					title : updatedInfo.title,
+					image : updatedInfo.image,
+					article : updatedInfo.content,
+					published : updatedInfo.published
+				})
+				.expect(200)
+
+		)
+		it('should return the right response', async () =>
+			fixtures.api()
+				.patch(`/api/v1/articles/${article.id}`)
+				.set('Authorization', `Bearer ${accessToken}`)
+				.send({ 
+					title : updatedInfo.title,
+					image : updatedInfo.image,
+					article : updatedInfo.content,
+					published : updatedInfo.published
+				})
+				.expect(200)
+				.then((res) => {
+					expect(res.body.data.userId).to.be.an('number')
+					expect(res.body.data.articleId).to.be.an('number')
+					expect(res.body.data.createdOn).to.be.a('string')
+					delete res.body.data.userId
+					delete res.body.data.articleId
+					delete res.body.data.createdOn
```

Comment:
we deleted the other data in the test because they were values we couldn't have known at that point

in that case, we were creating an article, so we won't know the id it comes with, neither would we know the  exact `createdOn` hence the deletion

Reply:
yeah, that makes sense